### DISCUSSION TITLE
TASK SCHEDULER Argument Invalid

### DISCUSSION THREAD

**Discussion by JazzyJazzjazzy** [T-Troll/alienfx-tools#282] [General]
Hello, I have created a task for starting AlienFans GUI.exe upon log-in. I attempted to create the same task for AlienFX and receive errors. I exported the working on for AlienFans and changed the path to the program and the task name but it continues to say "Argument not Valid". AlienFx is set to run with Administrator rights. Any ideas?
[AlienFans Start xml.txt](https://github.com/T-Troll/alienfx-tools/files/10671723/AlienFans.Start.xml.txt)

**Comment by T-Troll** [comment]
1. It's quite useless to run both `AlienFX-GUI` and `AlienFan-GUI` at the same time - AlienFX include all Fan functionality and share the same configuration.
2. Yes, they use different start settings. Just switch "Start with windows" in settings.

**Reply by JazzyJazzjazzy** [reply]
Agreed, but I like the functionality of hovering over the Fan icon to see fan speeds. otherwise, I have to open AlienFx.

**Reply by JazzyJazzjazzy** [reply]
Also, I thought I had seen mention in your notes regarding delaying the start up for some reason after starting windows. For this reason, I used Task Manager to delay its startup.

**Reply by T-Troll** [reply]
Delay is only needed in case you use Libra Hardware Monitor (to wait until it start).

Anyway, i don't recommend using both Fan GUI and FX GUI at the same time for these reasons:
- CPU load and fan control mess. They both run monitoring function, so it checks sensors and set fans continuously. In case both run, CPU usage x2. Even worse, Fan GUI always use system fan curves, but FX GUI can use fan curves from active profile. In case it is different, both will set boost values, so fans will dizzy between it.
- Configuration mess. They both share the same system fans configuration and curves. And they both save it at exit. So, in case you change something in one app (even if you press "Save" after it), you have 50% probability to lose the changes after system restart (in case app you change it quit faster, then other).

Being honest, Fan GUI is mostly for people doesn't have lights (so G's doesn't) and want to keep it simple.

Ok, I got your point about tooltip, let me check about adding this data to FX GUI as well (not so easy, because it's not update it so frequently now).

**Reply by JazzyJazzjazzy** [reply]
Thank you for the detailed description. I will remove AlienFx from start up because Fan Speed is more important to me and I rarely change light configuration.

**Reply by T-Troll** [reply]
Better keep FX - it can do more even for fans only (f.e. different profiles for different situations).
I'll try to add fans to tooltip at next release.

**Reply by JazzyJazzjazzy** [reply]
TOOL TIPS are a Great and Welcome Addition, thank you, sir.

**Reply by T-Troll** [reply]
Thanks. F*cking autocorrection....

**Reply by JazzyJazzjazzy** [reply]
Just thought you might like to know that first three Auto Starts of AlienFx Control resulted in BSODs. Watch Dog violation was reason given. It has settled down now.

Also, upon Start Up, I have two identical AlienFx Control icons. One subsequently drops off after about 10 seconds.

Question: is there a server being called upon by your files in the DPTF folder or are they inactive unless we move them to proper folder?

Thank you.

**Reply by T-Troll** [reply]
BSOD? Are you using Fan_V1 version?
I'll check about icons, but seems like you start it 2 times! Check Task Scheduler, BUT FIRST CHECK OUT "Start with Windows" into "Settings"!
For latest versions, this scripts only need one time - if names retrieved, it's not used anymore, as well as DPTF call. Open RegEdit, navigate to HKCU/Software/AlienFan - you will see "DPTF" option. In case it is 0 - test done and not called again, in case it's 1 - still not get names.

**Reply by JazzyJazzjazzy** [reply]
I am not using Version 1.I deleted the Auto Start task and unchecked Start with Windows option. Then Re-Checked Start with Windows option, verified only one line for AlienFx in Task Scheduler. Suspect it is related to Start Minimize option being checked. I deleted the DPTF folder and set the registry key to "0".

**Reply by T-Troll** [reply]
No, "Start minimized" only switch window open state then app starts. It can't do like this.
In fact, V2 app doesn't use any low-level access now, so BSOD is quite wonderful for me. Are you sure it's AFX-initiated?
Hmm... Just as crazy idea - run `alienfx-cli probe` (answer "n" for question) and check light devices list - is it correct? No ghost entry?

**Reply by JazzyJazzjazzy** [reply]
The reason I asked about the server is that I had 89 DCOM calls in Event Viewer. I have DCOM disabled for security.

**Reply by T-Troll** [reply]
It's not DCOM, in fact, but WMI. Fan control and monitoring use it.

**Reply by JazzyJazzjazzy** [reply]
> No, "Start minimized" only switch window open state then app starts. It can't do like this. In fact, V2 app doesn't use any low-level access now, so BSOD is quite wonderful for me. Are you sure it's AFX-initiated? Hmm... Just as crazy idea - run `alienfx-cli probe` (answer "n" for question) and check light devices list - is it correct? No ghost entry?


C:\Program Files\AlienFX Tools>alienfx-cli probe
alienfx-cli v7.10.4
Dell API not found, 2 low-level devices found.
===== Device VID_0d62, PID_1a1c =====
+++++ Detected as: Darfon, Keyboard, APIv5 +++++
===== Device VID_187c, PID_0550 =====
+++++ Detected as: Alienware, Desktop/Notebook, APIv4 +++++
Do you want to set devices and lights names?n
Done.
C:\Program Files\AlienFX Tools>

**Reply by JazzyJazzjazzy** [reply]
> It's not DCOM, in fact, but WMI. Fan control and monitoring use it.

With DCOM disabled, are my readings accurate? How does this impact your program?

**Reply by T-Troll** [reply]
WMI Provider try to access DCOM at connection start for (possible) remote operation, but works without it. So no issues in operation.

**Comment by T-Troll** [comment]
> I deleted the DPTF folder

Which one? App or Windows? If second - here we are. Reinstall Intel Dynamic Tuning driver ASAP!

**Reply by JazzyJazzjazzy** [reply]
The app folder. Not Windows.

**Comment by JazzyJazzjazzy** [comment]
https://user-images.githubusercontent.com/40358464/218016165-17c500d2-10c9-4193-9a80-409d94057978.mp4

**Comment by JazzyJazzjazzy** [comment]
Will not stop flashing

**Reply by T-Troll** [reply]
Oops! Bug. Thanks!

**Reply by T-Troll** [reply]
I update latest release, should fix this.

**Comment by JazzyJazzjazzy** [comment]
Sorry for large file.

**Comment by JazzyJazzjazzy** [comment]
![Screenshot_20230210-174230-868](https://user-images.githubusercontent.com/40358464/218072424-5d75e75b-6af5-4c7c-ad07-39b2465072cc.png)

**Comment by JazzyJazzjazzy** [comment]
![Event Viewer](https://user-images.githubusercontent.com/40358464/218072567-82a29f1d-f691-453d-892a-b5a254adfa72.png)

**Comment by JazzyJazzjazzy** [comment]
Once I HOVER over one of the icons, it disappears. No extra icon observed when starting program from Start Menu versus using Auto Start.
Also got BSOD on first restart. 
Update Flashing is gone. Thank you.

**Reply by T-Troll** [reply]
I check clean start, no issues.
In general, ghost icon can appear in 2 cases:
- App crashes
- App elevated to admin (manually, with UAC dialog) - BTW, i'll fix this at next release.

Please check event log about GUI app crashes.

**Reply by JazzyJazzjazzy** [reply]
UAC is disabled and program is not set to Run as Administrator. There are no GUI app crashes listed in Event Manager.

"No extra icon observed when starting program from Start Menu versus using Auto Start." is incorrect. I do have Ghost Icon when using Start Menu after boot up. Thanks.

**Reply by T-Troll** [reply]
Do we're still talking about `alienfx-gui`? It can't act like this if running once. `alienfan-gui` can.

**Reply by JazzyJazzjazzy** [reply]
Hello, fixed Ghost Icon on Start Up by deleting all registry entries referring to AlienFx from regedit. Deleted the AlienFx folder from Program Files, rebooted. Works fine now.

Notes: Previously used .msi file to install AlienFx Tools. Never uninstalled it but Updated program by downloading .zip file and replacing program components in the C:/Program/AlienFx folder.

Checked after compete new install and found only 45 reg entries vs, the 90 reg entries that I deleted.

Question? Where is my Fan Curve info stored? Curve was retained even after complete uninstall.

**Reply by T-Troll** [reply]
I found the reason. It can happened then default profile have non-default light state (dimmed, off). Will fix into next release.
There are 4 different reg entry. Look at `alienfx-config` for details.

**Reply by JazzyJazzjazzy** [reply]
Understood, thank you.

**Reply by JazzyJazzjazzy** [reply]
Okay, I checked photo of previous install before complete uninstall. Dim Lights was checked. Upon reinstall, since deleted reg entries, assuming reg entry for Dim Lights, started in default with Dim Lights unchecked. Good stuff.

**Reply by T-Troll** [reply]
You just mention configuration cleaning helps. It shouldn't be so due to the way i handle icon... except this case. So i check and reproduce.
